File: sdks/dotnet/src/Nukleio/NukleioApiException.cs
namespace Nukleio;

/// <summary>An unsuccessful or malformed response from the Nukleio API.</summary>
public sealed class NukleioApiException : Exception
{
    public NukleioApiException(string message, int? statusCode = null, string? responseBody = null)
        : base(message)
    {
        StatusCode = statusCode;
        ResponseBody = responseBody;
    }

    public int? StatusCode { get; }
    public string? ResponseBody { get; }
}

File: sdks/java/src/main/java/io/github/ericnohara/nukleio/NukleioApiException.java
package io.github.ericnohara.nukleio;

/** An unsuccessful or malformed response from the Nukleio API. */
public class NukleioApiException extends RuntimeException {
    private final int statusCode;
    private final String responseBody;

    public NukleioApiException(String message, int statusCode, String responseBody) {
        super(message);
        this.statusCode = statusCode;
        this.responseBody = responseBody;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public String getResponseBody() {
        return responseBody;
    }
}
